subnet/config.go

```diff
@@ -53,16 +53,28 @@ func ParseConfig(s string) (*Config, error) {
 	}
 
 	if cfg.SubnetLen > 0 {
-		if cfg.SubnetLen < cfg.Network.PrefixLen {
-			return nil, errors.New("HostSubnet is larger network than Network")
+		// SubnetLen needs to allow for a tunnel and bridge device on each host.
+		if cfg.SubnetLen > 30 {
+			return nil, errors.New("SubnetLen must be less than /31")
+		}
+
+		// SubnetLen needs to fit _more_ than twice into the Network.
+		// the first subnet isn't used, so splitting into two one only provide one usable host.
+		if cfg.SubnetLen < cfg.Network.PrefixLen+2 {
+			return nil, errors.New("Network must be able to accommodate at least four subnets")
 		}
 	} else {
-		// try to give each host a /24 but if the whole network
-		// is /24 or smaller, half the network
+		// try to give each host a /24 but if the whole network is /24 or smaller, split the network into four.
+		// the first subnet isn't used, so splitting into two one only provide one usable host.
 		if cfg.Network.PrefixLen < 24 {
```

Comment:
shouldn't this be `<=` according to the comment?

Reply:
@gunjan5 I updated the comments and code on these lines. PTAL.